internal/support/scheduler/schedule.go

```diff
@@ -0,0 +1,482 @@
+	scheduleContext.ScheduleEventsMap[scheduleEventId] = scheduleEvent
+	scheduleEventIdToScheduleIdMap[scheduleEventId] = scheduleId
+}
+
+//endregion
+
+//region schedule CRUD
+func querySchedule(scheduleId string) (models.Schedule, error) {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	scheduleContext, exists := scheduleIdToContextMap[scheduleId]
+	if !exists {
+		LoggingClient.Warn("can not find a schedule context with schedule id : " + scheduleId)
+		return models.Schedule{}, nil
+	}
+
+	LoggingClient.Debug("querying found the schedule with id : " + scheduleId)
+
+	return scheduleContext.Schedule, nil
+}
+
+func addSchedule(schedule models.Schedule) error {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	scheduleId := schedule.Id.Hex()
+	LoggingClient.Debug("adding the schedule with id : " + scheduleId + " at time " + schedule.Start)
+
+	if _, exists := scheduleIdToContextMap[scheduleId]; exists {
+		LoggingClient.Warn("the schedule context with id " + scheduleId + " already exist ")
+		return nil
+	}
+
+	context := ScheduleContext{
+		ScheduleEventsMap: make(map[string]models.ScheduleEvent),
+		MarkedDeleted:     false,
+	}
+
+	LoggingClient.Debug("resetting the schedule with id : " + scheduleId)
+	context.Reset(schedule)
+
+	addScheduleOperation(schedule, &context)
+
+	LoggingClient.Debug("added the schedule with id : " + scheduleId)
+
+	return nil
+}
+
+func updateSchedule(schedule models.Schedule) error {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	LoggingClient.Debug("updating the schedule with id : " + schedule.Id.Hex())
+
+	scheduleId := schedule.Id.Hex()
+	context, exists := scheduleIdToContextMap[scheduleId]
+	if !exists {
+		LoggingClient.Error("the schedule context with id " + scheduleId + " does not exist ")
+		return errors.New("the schedule context with id " + scheduleId + " does not exist ")
+	}
+
+	LoggingClient.Debug("resetting the schedule with id " + scheduleId)
+	context.Reset(schedule)
+
+	LoggingClient.Debug("updated the schedule with id : " + scheduleId)
+
+	return nil
+}
+
+func removeSchedule(scheduleId string) error {
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	LoggingClient.Debug("removing the schedule with id : " + scheduleId)
+
+	scheduleContext, exists := scheduleIdToContextMap[scheduleId]
+	if !exists {
+		LoggingClient.Error("can not find schedule context with schedule id : " + scheduleId)
+		return errors.New("can not find schedule context with schedule id : " + scheduleId)
+	}
+
+	LoggingClient.Debug("removing all the mappings of schedule event id to schedule id : " + scheduleId)
+	for eventId := range scheduleContext.ScheduleEventsMap {
+		delete(scheduleEventIdToScheduleIdMap, eventId)
+	}
+
+	deleteScheduleOperation(scheduleId, scheduleContext)
+
+	LoggingClient.Debug("removed the schedule with id : " + scheduleId)
+
+	return nil
+}
+
+//endregion
+
+//region schedule event CRUD
+func queryScheduleEvent(scheduleEventId string) (models.ScheduleEvent, error) {
+	mutex.Lock()
```

Comment:
GoLand currently identifies this as an unused function. If so, can be removed?

Reply:
were now using this.